L1Trigger/L1TTwinMux/src/DTRPCBxCorrection.cc

```diff
@@ -44,122 +45,140 @@ void DTRPCBxCorrection::run( const edm::EventSetup& c) {
-	 		if(dtts && dtts->code()<m_QualityLimit && (!dttsnew || shifted[final_bx+3] || dups[final_bx+3])) {
-		 		dtts_sh = new L1MuDTChambPhDigi( final_bx , dtts->whNum(), dtts->scNum(), dtts->stNum(),dtts->phi(), dtts->phiB(), dtts->code(), dtts->Ts2Tag(), dtts->BxCnt(),1);
-		 		l1ttma_outsh.push_back(*dtts_sh);
-		 		shifted[init_bx+3] = true;
-		 		shift_1 = true;
-	 		}
-	 		if(dtts && dtts->code()<m_QualityLimit && dttsnew  ) dups[init_bx+3] = true;
-
-			///dtts exists and qual lt m_QualityLimit and dttsnew exists and the previous (shift_1) prim was not shifted and there is empty space in second TS
-	 		if(dtts && dtts->code()<m_QualityLimit && dttsnew && !shift_1 && !m_phiDTDigis.chPhiSegm(wheel,station,sector,final_bx,flipBit(track_seg)) ) {
-				///XXX: Source of discrepancies
-				///in order to send as second TS the two prims must come from different halves of the station
-				///this information does not exist in data
-				///'simulate' this information by requiring different sign in phis and dphi>100
-				 if(sign(dtts->phi())!=sign(dttsnew->phi())&& deltaPhi(-(dtts->phi()),dttsnew->phi())>100) {
-					 dtts_sh2nd  = new L1MuDTChambPhDigi( final_bx , dtts->whNum(), dtts->scNum(), dtts->stNum(),dtts->phi(), dtts->phiB(), dtts->code(), flipBit(track_seg), dtts->BxCnt(),1);
-
-					 secondTs[final_bx+3] = true;
-					 dups[init_bx+3] = false;
-					 shifted[init_bx+3] = true;
-				 }
-	 	 	}
-     shiftedPhiDTDigis.setContainer(l1ttma_outsh);
+       L1MuDTChambPhDigi *dtts_sh = 0;
+       std::vector<L1MuDTChambPhDigi> l1ttma_outsh;
+       //Find the pair the min dphi(rpc,dt)
+       unsigned int min_index = std::distance(delta.begin(), std::min_element(delta.begin(), delta.end())) + 0;
+       int init_bx = 0, final_bx = 0;
+
+       if ( ((delta_0.size() <= min_index) && ( min_index < (delta_0.size() + delta_p.size()) ) && delta_p.size()!=0 ) ) {
+           init_bx = ibx_dtp;
+           final_bx = fbx_dtp;
+           }
+       else if ( (delta_0.size() + delta_p.size()) <= min_index  && delta_m.size()!=0  ) {
+           init_bx = ibx_dtm;
+           final_bx = fbx_dtm;
+          }
+       else continue;
+        //Primitve to be shifted in place of dttsnew
+       if (m_phiDTDigis_tm->chPhiSegm(wheel,station,sector,init_bx ,track_seg))
+         dtts.reset(new L1MuDTChambPhDigi(init_bx,m_phiDTDigis_tm->chPhiSegm(wheel,station,sector,bx,track_seg)->whNum(),m_phiDTDigis_tm->chPhiSegm(wheel,station,sector,bx,track_seg)->scNum(),m_phiDTDigis_tm->chPhiSegm(wheel,station,sector,bx,track_seg)->stNum(),m_phiDTDigis_tm->chPhiSegm(wheel,station,sector,bx,track_seg)->phi(),m_phiDTDigis_tm->chPhiSegm(wheel,station,sector,bx,track_seg)->phiB(),m_phiDTDigis_tm->chPhiSegm(wheel,station,sector,bx,track_seg)->code(),m_phiDTDigis_tm->chPhiSegm(wheel,station,sector,bx,track_seg)->Ts2Tag(),m_phiDTDigis_tm->chPhiSegm(wheel,station,sector,bx,track_seg)->BxCnt(),m_phiDTDigis_tm->chPhiSegm(wheel,station,sector,bx,track_seg)->RpcBit()));
```

Comment:
This will probably give a bad performance since new L1MuDTChampPhDigis are constructed all the time.
Better to keep the raw pointers in this case.